ListLibrary/MyList.cs

```diff
@@ -0,0 +1,519 @@
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            int result = 0;
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(_items[result]) == 1)
+                {
+                    result = i;
+                }
+            }
+
+            return result;
+        }
+
+        public int IndexOfMin()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            int result = 0;
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(_items[result]) == -1)
+                {
+                    result = i;
+                }
+            }
+
+            return result;
+        }
+
+        public void SortByDesc()
+        {
+            T x;
+            int j;
+
+            for (int i = 1; i < _size; i++)
+            {
+                x = _items[i];
+                j = i;
+                while (j > 0 && _items[j - 1].CompareTo(x) == -1)
+                {
+                    Swap(ref _items[j], ref _items[j - 1]);
+                    j -= 1;
+                }
+
+                _items[j] = x;
+            }
+        }
+
+        public void SortByAsc()
+        {
+            for (int i = 0; i < _size - 1; i++)
+            {
+                int min = i;
+
+                for (int j = i + 1; j < _size; j++)
+                {
+
+                    if (_items[j].CompareTo(_items[min]) == -1)
+                    {
+                        min = j;
+                    }
+                }
+
+                Swap(ref _items[i], ref _items[min]);
+            }
+        }
+
+        #endregion
+
+        #region Other
+
+        public void Reverse()
+        {
+            for (int i = 0; i < _size / 2; i++)
+            {
+                Swap(ref _items[i], ref _items[_size - 1 - i]);
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= _size || index < 0)
+                {
+                    throw new ArgumentOutOfRangeException("Index is out of range");
+                }
+                return _items[index];
+            }
+
+            set
+            {
+                if (index >= _size || index < 0)
+                {
+                    throw new ArgumentOutOfRangeException("Index is out of range" + _size);
+                }
+                _items[index] = value;
+            }
+        }
+
+        #endregion
+
+        private void UpdateCapacity(int min)
+        {
+            if (min < _size)
+            {
+                throw new ArgumentOutOfRangeException("Capacity is too small");
+            }
+
+            if (_items.Length < min)
+            {
+                int newCapacity = (int)(_items.Length * _capacityRise);
+
+                if (newCapacity > int.MaxValue)
```

Comment:
`int.MaxValue` check?)
It will never occurs because you will get overflow.
It should be checked with `checked` attribute if you want.

Reply:
fixed